### PR TITLE
Add option to retrieve iteration using iteration id

### PR DESCRIPTION
**Pull request by lewisdylan**
Added an option in the base controller to retrieve the iteration by the iteration id

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ application up and running.
 The project uses [rubocop](https://github.com/rubocop-hq/rubocop) for code linting.
 
 ```rb
-bundle exec rubocop
+
```
Comment (bumi) [inline README.md]: why did you delete this?

File: app/controllers/api/answers_controller.rb [hunk 2]
```diff
@@ -8,7 +8,8 @@ class AnswersController < Api::BaseController
     def create
       answer = Answer.find(request_params[:answer_id])
       if current_iteration.answer_question(answer)
-        render json: true
+        question = Question.where(id: answer.next_question_id).first
+        render json: question.id
```
Comment (bumi) [inline app/controllers/api/answers_controller.rb]: that should be `answer.next_question` - the rails relation is doing the lookup.

File: app/controllers/api/base_controller.rb [hunk 3]
```diff
@@ -19,7 +21,7 @@ def require_current_user
     end
 
     def require_current_iteration
-      render json: { error: 'Iteration not found' }, status: :unauthorized if current_user.blank? || iteration.blank?
+      render json: { error: 'Iteration not found' }, status: :unauthorized if current_user.blank? || current_iteration.blank?
```
Comment (bumi) [inline app/controllers/api/base_controller.rb]: :+1: thanks for fixing this.

File: app/controllers/api/base_controller.rb [hunk 4]
```diff
@@ -11,6 +11,8 @@ def current_user
     # the current iteration is always the last iteration
     # When a user creates a new iterations the old ones can not be changed again
     def current_iteration
+      return @current_iteration ||= current_user.iterations.find(params[:iteration_id]) || current_user.iterations.last if params[:iteration_id]
```
Comment (bumi) [inline app/controllers/api/base_controller.rb]: hmm. I need to think about this.

File: README.md (hunk 1 above)
Comment (lewisdylan) [reply README.md]: Sorry it was mistake

File: app/controllers/api/answers_controller.rb (hunk 2 above)
Comment (lewisdylan) [reply app/controllers/api/answers_controller.rb]: ooh yeah that's true
